### PR TITLE
Feature/Test: User Authentication and Tests Using RSpec

### PR DESCRIPTION
**Pull request by sal-wtag**
This Pull Request introduces the User Authentication feature and its corresponding tests using `RSpec`.

## Description

This PR includes the implementation of user authentication using JSON Web Tokens (JWT). Users can log in using their email and password, upon which the server provides a JWT in a cookie. For subsequent requests, the cookie-stored JWT is used to verify the user. When a user logs out, the cookie is cleared.

### `app/services/jwt_service.rb`

This file contains a simple service for generating, encoding, verifying and decoding JWTs.

### `app/controllers/auth_controller.rb`

This file includes the `AuthController` which handles the authentication requests. It includes methods for logging in and logging out users.

### `app/controllers/concerns/authenticable.rb`

This file includes the `Authenticable` module which is included in `AuthController`, `UsersController`, and `ConfirmationsController`. It provides methods for authenticating users and managing the current user session.

### `app/views/auth`

This directory contains necessary views (e.g. forms) for authentication

## Demo

[user-auth-login-logout.webm](https://github.com/samin-al-wasee/medical-appointment-scheduler/assets/167951273/da78c1e4-1916-4b7e-b53e-79c829f7009d)

## Testing

The new feature has been tested using RSpec. The tests ensure that the user authentication works as expected and handles edge cases. Check `app/spec/controllers/auth_controller_spec.rb`.

### Results

<pre>
AuthController
  GET #new when not authenticated
✓ should return success response
✓ should render the new template
  GET #new when already authenticated
✓ should redirect to root_url
  POST #create with invalid email
✓ should return alert message
✓ should redirect to new_auth_url
  POST #create with invalid password
✓ should return alert message
✓ should redirect to new_auth_url
  POST #create with inactive user
✓ should return alert message
✓ should redirect to new_auth_url
  POST #create with valid email, password and active user
DEPRECATION WARNING: `Rails.application.secrets` is deprecated in favor of `Rails.application.credentials` and will be removed in Rails 7.2. (called from <class:JwtService> at /home/wasee/medical-appointment-scheduler/app/services/jwt_service.rb:4)
✓ should set current_user
✓ should set jwt cookie
✓ should redirect to root_url
  POST #create when already authenticated
✓ should redirect to root_url
  DELETE #destroy when authenticated
✓ should delete current_user
✓ should delete jwt cookie
✓ should redirect to new_auth_url
  DELETE #destroy when not authenticated
✓ should return alert message
✓ should redirect to new_auth_url

Finished in 0.63654 seconds (files took 2.17 seconds to load)
18 examples, 0 failures
</pre>

### CODE REVIEW

File: app/controllers/auth_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
```
Comment (kth-wtag) [inline app/controllers/auth_controller.rb:1]: remove it

File: config/application.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
-require_relative "boot"
+# frozen_string_literal: true
```
Comment (kth-wtag) [inline config/application.rb:1]: ditto
